lib/doorkeeper/client_authentication/fallback_mechanism.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module Doorkeeper
+  module ClientAuthentication
+    class FallbackMechanism
+      def authenticate(request)
+        Doorkeeper::ClientAuthentication::Credentials.new(nil, nil)
```

Comment:
I think this should actually just be returning nil, not an instance of Credentials, which is why the `lib/doorkeeper/oauth/token_introspection.rb` condition failed.